libcontainer/nsenter/idmap.h

```diff
@@ -0,0 +1,78 @@
+#include <sys/mount.h>
+// Centos-7 doesn't have this file nor the __has_include() directive, so let's
+// just leave this commented out and we can uncomment when it hits EOL (2024-06-30).
+//#include <linux/mount.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+/* mount_setattr() */
+#ifndef MOUNT_ATTR_IDMAP
+#define MOUNT_ATTR_IDMAP 0x00100000
+#endif
+
+#ifndef __NR_mount_setattr
+	#if defined __alpha__
+		#define __NR_mount_setattr 552
+	#elif defined _MIPS_SIM
+		#if _MIPS_SIM == _MIPS_SIM_ABI32	/* o32 */
+			#define __NR_mount_setattr (442 + 4000)
+		#endif
+		#if _MIPS_SIM == _MIPS_SIM_NABI32	/* n32 */
+			#define __NR_mount_setattr (442 + 6000)
+		#endif
+		#if _MIPS_SIM == _MIPS_SIM_ABI64	/* n64 */
+			#define __NR_mount_setattr (442 + 5000)
+		#endif
+	#elif defined __ia64__
+		#define __NR_mount_setattr (442 + 1024)
+	#else
+		#define __NR_mount_setattr 442
+	#endif
```

Comment:
nit: Go won't support Alpha and IA64

Reply:
@AkihiroSuda I removed it. But now it is `if mips <mips> else #define __NR_mount_setattr 442`, which can be tricky to debug if go adds another arch that has a different syscall number.

Given your comment and that adding support for alpha or ia64 seems unlikely, This is probably safe. But let me know if you change your mind now that it looks like how I mentioned :)